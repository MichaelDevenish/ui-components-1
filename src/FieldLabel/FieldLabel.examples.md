# Field Label

## Basic Field Label

```jsx
  <div>
    <FieldLabel>
      Hello boys!
    </FieldLabel>
    <TextInput name='example-input1' />
  </div>
```

## Field Label with `htmlFor`

```jsx
  <div>
    <FieldLabel
      htmlFor='example-input2'
    >
      Hello ladies!
    </FieldLabel>
    <TextInput name='example-input2' />
  </div>
```

## Field Label with isInvalid

```jsx
  <div>
    <FieldLabel
      isInvalid
    >
      Gentlemen...
    </FieldLabel>
    <TextInput name='example-input3' isInvalid />
  </div>
```
