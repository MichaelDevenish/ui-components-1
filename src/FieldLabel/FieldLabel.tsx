import React from 'react'
import { StyledFieldLabel } from './style'
import classNames from 'classnames'

export interface FieldLabelProps {
  /** ID of input to attach to */
  htmlFor?: string
  /** If true, adds invalid label class to component */
  isInvalid?: boolean
}

export class FieldLabel extends React.PureComponent<FieldLabelProps> {
  public render (): JSX.Element {
    const {
      htmlFor,
      children,
      isInvalid
    } = this.props

    return (
      <StyledFieldLabel
        htmlFor={htmlFor}
        className={classNames(
          { 'is-invalid-label': isInvalid }
        )}
      >
        {children}
      </StyledFieldLabel>
    )
  }
}
