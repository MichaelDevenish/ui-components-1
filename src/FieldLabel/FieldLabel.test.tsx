import { mount } from 'enzyme'
import React from 'react'

import { FieldLabel } from './FieldLabel'

describe('<FieldLabel />', () => {
  it(`should render a label`, () => {
    const wrapper = mount(
      <FieldLabel>
        Hello boys.
      </FieldLabel>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it(`should render a label with a 'for' attribute`, () => {
    const wrapper = mount(
      <FieldLabel
      htmlFor='input-id'
      >
        Hello ladies.
      </FieldLabel>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it(`should render a red label if isInvalid is true`, () => {
    const wrapper = mount(
      <FieldLabel
        isInvalid
      >
        Hello ladies.
      </FieldLabel>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
