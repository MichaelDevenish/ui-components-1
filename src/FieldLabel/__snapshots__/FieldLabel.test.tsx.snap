// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldLabel /> should render a label 1`] = `
.c0 {
  color: hsl(210,13.4%,47.2%);
  font-size: 0.9375rem;
  line-height: 1.8;
}

.c0.is-invalid-label {
  color: rgb(255,77,80);
}

<FieldLabel>
  <styled.label
    className=""
  >
    <label
      className="c0"
    >
      Hello boys.
    </label>
  </styled.label>
</FieldLabel>
`;

exports[`<FieldLabel /> should render a label with a 'for' attribute 1`] = `
.c0 {
  color: hsl(210,13.4%,47.2%);
  font-size: 0.9375rem;
  line-height: 1.8;
}

.c0.is-invalid-label {
  color: rgb(255,77,80);
}

<FieldLabel
  htmlFor="input-id"
>
  <styled.label
    className=""
    htmlFor="input-id"
  >
    <label
      className="c0"
      htmlFor="input-id"
    >
      Hello ladies.
    </label>
  </styled.label>
</FieldLabel>
`;

exports[`<FieldLabel /> should render a red label if isInvalid is true 1`] = `
.c0 {
  color: hsl(210,13.4%,47.2%);
  font-size: 0.9375rem;
  line-height: 1.8;
}

.c0.is-invalid-label {
  color: rgb(255,77,80);
}

<FieldLabel
  isInvalid={true}
>
  <styled.label
    className="is-invalid-label"
  >
    <label
      className="is-invalid-label c0"
    >
      Hello ladies.
    </label>
  </styled.label>
</FieldLabel>
`;
